addons/website_event/models/event.py

```diff
@@ -115,3 +134,50 @@ def action_open_badge_editor(self):
             'target': 'new',
             'url': '/report/html/%s/%s?enable_editor' % ('event.event_event_report_template_badge', self.id),
         }
+
+    @api.multi
+    def get_ics_file(self):
+        """ Returns iCalendar file for the event invitation.
+            :returns a dict of .ics file content for each event
+        """
+        result = {}
+        if not vobject:
+            return result
+
+        for event in self:
+            cal = vobject.iCalendar()
+            cal_event = cal.add('vevent')
+
+            if not event.date_begin or not event.date_end:
+                raise UserError(_("First you have to specify the date of the invitation."))
+            cal_event.add('created').value = fields.Datetime.from_string(fields.Datetime.now()).replace(tzinfo=pytz.timezone('UTC'))
+            cal_event.add('dtstart').value = fields.Datetime.from_string(event.date_begin).replace(tzinfo=pytz.timezone('UTC'))
+            cal_event.add('dtend').value = fields.Datetime.from_string(event.date_end).replace(tzinfo=pytz.timezone('UTC'))
+            cal_event.add('summary').value = event.name
+            if event.address_id:
+                cal_event.add('location').value = event.sudo().address_id.contact_address
+
+            attendee_ids = self.env.context.get('attendee_ids')
+            attendees = self.env['event.registration'].sudo().browse(attendee_ids)
```

Comment:
This is a security hole. Basically you allow people to browse whatever attendees in sudo using context.